mne/preprocessing/eog.py

```diff
@@ -249,3 +250,112 @@ def create_eog_epochs(raw, ch_name=None, event_id=998, picks=None, tmin=-0.5,
                         reject_by_annotation=reject_by_annotation,
                         decim=decim)
     return eog_epochs
+
+
+def eog_regression(inst, eog_evokeds=None, eog_channels=None, picks=None):
+    """Remove EOG signals from the EEG channels by regression.
+
+    Employs linear regression to remove EOG signals from other channels, as
+    described in [1]_. Optionally, one may chose to use evoked blink/saccade
+    data to obtain the regression weights as described in [2]_.
+
+    The operation is performed in-place.
+
+    Parameters
+    ----------
+    inst : Raw | Epochs
+        The data on which the EOG correction produce should be performed.
+    eog_evokeds : Evoked | list of Evoked | None
+        Optional recordings of averaged eye movements. For example averaged
+        blinks and/or saccades (use a list to supply different kinds of
+        movements). When specified, regression weights will be fitted on this
+        data, before being applied to the data given as ``inst`` parameter.
+        Regression weights obtained from averaged data might be more robust
+        [2]_.
+    eog_channels : str | list of str | None
+        The names of the EOG channels to use in the regression. By default, all
+        EOG channels are used.
+    picks : list of int | None
+        Channels from which to remove the EOG. By default, the correction is
+        applied to EEG channels only.
```

Comment:
It seems like it should be applied to all data channels by default, or at least EEG+MEG. But it's easier to just say "all data channels" and use the key